internal/utils/utils_test.go

```diff
@@ -203,3 +205,14 @@ func TestBuildImagePath(t *testing.T) {
 		})
 	}
 }
+
+func TestIsInHelmMode(t *testing.T) {
+	orig := os.Getenv(CLIModeVariableName)
+	assert.NoError(t, os.Setenv(CLIModeVariableName, "helm"))
+	assert.True(t, IsInHelmMode())
+	assert.NoError(t, os.Setenv(CLIModeVariableName, "classic"))
+	assert.False(t, IsInHelmMode())
+	assert.NoError(t, os.Setenv(CLIModeVariableName, "random"))
+	assert.False(t, IsInHelmMode())
+	assert.NoError(t, os.Setenv(CLIModeVariableName, orig))
```

Comment:
Let's move the setting of the original value to a `defer` func such that is reset also in case of an error or panic:

```suggestion
	orig := os.Getenv(CLIModeVariableName)
	defer func() {
		assert.NoError(t, os.Setenv(CLIModeVariableName, orig))
	}()
	assert.NoError(t, os.Setenv(CLIModeVariableName, "helm"))
	assert.True(t, IsInHelmMode())
	assert.NoError(t, os.Setenv(CLIModeVariableName, "classic"))
	assert.False(t, IsInHelmMode())
	assert.NoError(t, os.Setenv(CLIModeVariableName, "random"))
	assert.False(t, IsInHelmMode())
```

Reply:
https://github.com/cilium/cilium-cli/pull/1612 😎